### ISSUE TITLE
Succescriteria bij component Button

### ISSUE THREAD

**Issue by MarjonBakker** [nl-design-system/documentatie#1392]
Gerelateerd #1375 

## Notities Marjon:
- Kopje 'WCAG 2.2' veranderd in 'Succescriteria WCAG 2.2'. Het woord 'Succescriterium' weggelaten uit de lijst, zodat je als screenreadergebruiker de unieke tekst per SC meteen hoort.
- `<h3 role="presentation" class="utrecht-heading-3">Succescriterium 2.1.3</h3>` veranderd in `h4` zonder role=presentation, omdat een heading level 4 hier nut heeft.
  - in `TaskList.tsx`: verander `<Heading3 role="presentation">{title}</Heading3>` in `<Heading4 >{title}</Heading4>`
- 2.1.3 vervangen door 2.1.1: https://codefornl.slack.com/archives/C02CWGL2M18/p1726469245214309
- Ik heb eigen teksten gebruikt in plaats van de summary's, omdat ik vond dat de standaarteksten niet altijd precies genoeg waren voor dit specifieke component.

## Discussiepunten

1.  Vaak hebben criteria met elkaar te maken, zoals 1.1.1 Niet-tekstuele content met 2.4.6 Koppen en labels en 4.1.2 Naam, rol, waarde als het gaat om een alt-tekst die ook de accName van iets vormt. Willen we in zo'n geval kruisverwijzen? Nog zo'n setje: 1.4.11 Contrast van niet-tekstuele content, 2.4.7 Focus zichtbaar, 2.4.13 Focusweergave
2. Hoe groot maken we de scope hier? Bijvoorbeeld 3.2.4 Consistente identificatie, dat gaat om meerdere buttons op verschillende plaatsen.

#### [1.1.1 Niet-tekstuele content](https://www.nldesignsystem.nl/wcag/1.1.1)
Als het label van een button uit een afbeelding bestaat, dan heeft deze afbeelding een goed tekstalternatief, zodat de button een naam heeft die duidelijk maakt waar de button voor dient.

NL Design System richtlijnen:
* [Afbeeldingen als buttons](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/afbeelding-als-button)
* [De toegankelijke naam van een button](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/toegankelijke-naam)

#### [1.4.3 Contrast (minimum)](https://www.nldesignsystem.nl/wcag/1.4.3/)
Het contrast van de buttontekst met de achtergrondkleur is hoog genoeg. Dit is te controleren met de [Contrast checker](https://nldesignsystem.nl/contrast/).

NL Design System richtlijnen:
* [Geef tekst voldoende kleurcontrast](https://www.nldesignsystem.nl/richtlijnen/formulieren/visueel-ontwerp/tekst-goed-zichtbaar/)

#### [1.4.4 Herschalen van tekst](https://www.nldesignsystem.nl/wcag/1.4.4/)
Als je de tekst vergroot tot 200% (via browserzoom en via de browserinstellingen voor tekstgrootte) blijft de tekst in zijn geheel zichtbaar.

#### [1.4.5 Afbeeldingen van tekst](https://www.nldesignsystem.nl/wcag/1.4.5/)
Het label van de button bestaat uit gewone tekst, niet uit een afbeelding van tekst. Tenzij om een logo gaat, zoals het DigiD-logo.

NL Design System richtlijnen:
* [Afbeeldingen als buttons](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/afbeelding-als-button)

#### [1.4.11 Contrast van niet-tekstuele content](https://www.nldesignsystem.nl/wcag/1.4.11/)
Als het label van de button niet uit tekst bestaat maar uit bijvoorbeeld een icoon (zoals bij een zoekknop met een vergrootglas), dan is het contrast tussen het icoon en de achtergrond minimaal 3:1. Dit is te controleren met de [Contrast checker](https://nldesignsystem.nl/contrast/).

#### [1.4.12 Tekstafstand](https://www.nldesignsystem.nl/wcag/1.4.12/)
Als je de tekstafstand vergroot blijft de tekst in zijn geheel zichtbaar. Dit gaat om afstand tussen letters, woorden, regels en alinea's.

#### [2.1.1 Toetsenbord](https://www.nldesignsystem.nl/wcag/2.1.1/)
Je kunt de button focussen met de tabtoets en activeren met de spacebar en de entertoets.

NL Design System richtlijnen:
* [Toetsenbordbediening van een button](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/toetsenbordbediening)
* [Disabled submitbuttons](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/disabled-submitbuttons/)

#### [2.4.6 Koppen en labels](https://www.nldesignsystem.nl/wcag/2.4.6/)
Het label van de button maakt kort en bondig duidelijk waar de button voor dient.

NL Design System richtlijnen:
* [Duidelijke buttontekst die beschrijft wat de button doet](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/duidelijk-buttontekst/)

#### [2.4.7 Focus zichtbaar](https://www.nldesignsystem.nl/wcag/2.4.7/)
Wanneer een button de toetsenbordfocus krijgt is de focus zichtbaar.

NL Design System richtlijnen:
* [Maak toetsenbord focus goed zichtbaar](https://www.nldesignsystem.nl/richtlijnen/formulieren/visueel-ontwerp/focus-goed-zichtbaar/)

#### [2.4.13 Focusweergave](https://www.nldesignsystem.nl/wcag/2.4.13/)
Er is een goed zichtbare focusindicator die ook rekening houdt met het contrast met de verschillende achtergrondkleuren waarop de button gebruikt kan worden.

NL Design System richtlijnen:
* [Maak toetsenbord focus goed zichtbaar](https://www.nldesignsystem.nl/richtlijnen/formulieren/visueel-ontwerp/focus-goed-zichtbaar/)

#### [2.5.2 Aanwijzerannulering](https://www.nldesignsystem.nl/wcag/2.5.2)
Dit criterium maakt het moeilijker om een button onbedoeld te activeren met een muis, toetsenbord of touchscreen.

#### [2.5.3 Label in Naam](https://www.nldesignsystem.nl/wcag/2.5.3/)
De zichtbare naam van de button is gelijk aan de toegankelijke naam.

NL Design System richtlijnen:
* [De zichtbare naam moet overeenkomen met de toegankelijke naam](https://www.nldesignsystem.nl/richtlijnen/formulieren/labels/zichtbare-naam/)

#### [2.5.5 Grootte van het aanwijsgebied (uitgebreid)](https://www.nldesignsystem.nl/wcag/2.5.5/)
De button is groot genoeg om gemakkelijk aan te klikken (44 bij 44 pixels). Kleinere buttons staan niet te dicht op een andere button.

NL Design System richtlijnen:
* [Maak aanklikbare formuliervelden groot genoeg](https://www.nldesignsystem.nl/richtlijnen/formulieren/visueel-ontwerp/invoerveld-goed-aanklikbaar/)

#### [3.1.2 Taal van onderdelen](https://www.nldesignsystem.nl/wcag/3.1.2/)
Als een buttontekst in een andere taal is dan de taal van de pagina, dan heeft het element een `lang`-attribuut met de juiste taalcode.

Denk bijvoorbeeld aan buttons voor het veranderen van de taal van een pagina, bij meertalige websites.

#### [3.2.4 Consistente identificatie](https://www.nldesignsystem.nl/wcag/3.2.4/)
Buttons met gelijke functies hebben hetzelfde uiterlijk en hetzelfde label.

#### [4.1.2 Naam, rol, waarde](https://www.nldesignsystem.nl/wcag/4.1.2/)
De button heeft een rol van `button` en een toegankelijke naam die duidelijk maakt waar de button voor dient. Ook is de toestand (*state*) duidelijk.

NL Design System richtlijnen:
* [De toegankelijke naam van een button](https://www.nldesignsystem.nl/richtlijnen/formulieren/buttons/toegankelijke-naam)

**Comment by Robbert** [comment]
Ik zou graag wat nauwkeuriger ingaan op welke state duidelijk moet zijn 4.1.2. Het succescriterium gaat namelijk specifiek over dat states in te stellen moeten zijn, en dat er notifications zijn over wijzigingen van states. Er lijkt niet per se te staan dat states ook programatically determinable moeten zijn.

**Comment by Yolijn** [comment]
Mooie PR geworden! De enige vreemde 🦆 eend in de bijt is wat mij betreft de beschrijving bij criteria 2.5.2
> Dit criterium maakt het moeilijker om...

Bij de rest is het concreet wat het betekend en wat je moet doen, bij deze zou ik het nu nog niet weten.

**Comment by MarjonBakker** [comment]
> Mooie [PR](https://github.com/nl-design-system/documentatie/pull/1209) geworden, ik heb hem gemerged! De enige vreemde 🦆 eend in de bijt is wat mij betreft de beschrijving bij criteria 2.5.2
> 
> > Dit criterium maakt het moeilijker om...
> 
> Bij de rest is het concreet wat het betekend en wat je moet doen, bij deze zou ik het nu nog niet weten.

@Yolijn Klopt, goed gezien. Ik wilde deze acceptatiecriteria zo eenvoudig mogelijk houden, zonder technisch te worden. Technische vereisten kun je dan lezen in de [uitleg van het criterium](https://documentatie-git-docs-component-wcag-criteria-nl-design-system.vercel.app/wcag/2.5.2). Bij 2.5.2 vond ik dat heel lastig. We kunnen wel de tekst overnemen over up en down events?

**Comment by Yolijn** [comment]
> @Yolijn Klopt, goed gezien. Ik wilde deze acceptatiecriteria zo eenvoudig mogelijk houden, zonder technisch te worden. Technische vereisten kun je dan lezen in de [uitleg van het criterium](https://documentatie-git-docs-component-wcag-criteria-nl-design-system.vercel.app/wcag/2.5.2). Bij 2.5.2 vond ik dat heel lastig. We kunnen wel de tekst overnemen over up en down events?

Check! Zouden we niet de uitlegtekst van het criterium (deels) kunnen gebruiken? 
> Zorg ervoor dat, als de gebruiker een button indrukt met een aanwijzer zoals een muis of vinger, er de mogelijkheid is om actie te voorkomen of ongedaan te maken.
> 
> Dit voorkomt het per ongeluk aanraken en activeren van functies, waarvan het niet de bedoeling was.

**Comment by MarjonBakker** [comment]
> Ik zou graag wat nauwkeuriger ingaan op welke state duidelijk moet zijn 4.1.2. Het succescriterium gaat namelijk specifiek over dat states in te stellen moeten zijn, en dat er notifications zijn over wijzigingen van states. Er lijkt niet per se te staan dat states ook programatically determinable moeten zijn.

@Robbert Hoort dat hier thuis? Je kan doorklikken naar de [uitleg van 4.1.2](https://documentatie-git-docs-component-wcag-criteria-nl-design-system.vercel.app/wcag/4.1.2), daar zou volgens mij uitgebreidere uitleg moeten staan. Bij de acceptatiecriteria wil ik het graag zo eenvoudig mogelijk houden.

Of bedoel je de laatste zin te veranderen in bijvoorbeeld: "Ook is de toestand (state) duidelijk en door de gebruiker te veranderen met hulpsoftware"?

**Comment by Robbert** [comment]
Ik heb de zorg dat bij de meeste teksten ontwikkelaars en designers niet concreet geholpen zijn om een goede component te maken. De teksten lijken, eigenlijk net als WCAG, geschreven te zijn om als basis voor een beoordeling te maken.

Om mensen succesvol te laten zijn, dat er een combinatie van één of meer van de volgende aspecten nodig is:
- ik denk ik dat er in elk geval meer info nodig is over concrete oplossingen;
- of dat een explicietere verwijzing naar concrete info nodig is ("lees meer over welk HTML element je moet gebruiken in de volgende richtlijn";
- of misschien dat de teksten meer op makers dan op testers gericht moeten zijn. Bijv zoiets voor 1.4.4: "Gebruik nooit `px` voor `font-size` en `line-height`, maar gebruik bijvoorbeeld `rem`. Lees meer op ..."

**Comment by Robbert** [comment]
> Of bedoel je de laatste zin te veranderen in bijvoorbeeld: "Ook is de toestand (state) duidelijk en door de gebruiker te veranderen met hulpsoftware"?

Dat is inderdaad meer wat ik bedoelde.

**Comment by MarjonBakker** [comment]
@Yolijn Ja, dat is een idee, maar er moet niet teveel dubbelen. Ik stel voor: "Als de gebruiker een button indrukt met een aanwijzer zoals een muis of vinger, is er de mogelijkheid is om actie te voorkomen of ongedaan te maken."
